src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBadGenomicKmers.java

```diff
@@ -0,0 +1,135 @@
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceMultiSource;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import scala.Tuple2;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * SparkTool to identify 63-mers in the reference that occur more than 3 times.
+ *
+ * Created by tsharpe on 12/4/15.
+ */
+@CommandLineProgramProperties(summary="Use spark to find the set of high copy number kmers in a reference.",
+        oneLineSummary="find ref kmers with high copy number",
+        programGroup = SparkProgramGroup.class)
+public final class FindBadGenomicKmers extends GATKSparkTool
+{
+    private static final long serialVersionUID = 1L;
+    private static final Long MAX_KMER_FREQ = 3L;
+    private static final int KLEN = 63;
+    private static final int REF_REC_LEN = 10000;
+    private static final long CHUNK_SIZE = 3000000;
+
+    @Argument(doc = "file for ubiquitous kmer output", shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, optional = false)
+    private String outputFile;
+
+    @Override
+    public boolean requiresReference()
+    {
+        return true;
+    }
+
+    // Get the list of high copy number kmers in the reference, and write them to a file.
+    @Override
+    protected void runTool( final JavaSparkContext ctx )
+    {
+        SAMFileHeader hdr = getHeaderForReads();
+        SAMSequenceDictionary dict = null;
+        if ( hdr != null ) dict = hdr.getSequenceDictionary();
+        List<Kmer> killList = findBadGenomicKmers(ctx,getReference(),dict);
+        final File killFile = new File(outputFile);
+        try
+        {
+            final Writer writer = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(killFile)));
+
+            for ( Kmer kmer : killList )
+            {
+                writer.write(kmer.toString(KLEN));
+                writer.write('\n');
+            }
+            writer.close();
+        }
+        catch ( IOException e )
+        { throw new GATKException("Unable to write ubiquitous kmer kill file.", e); }
+    }
+
+    // Find high copy number kmers in the reference sequence
+    public static List<Kmer> findBadGenomicKmers( final JavaSparkContext ctx,
+                                                  final ReferenceMultiSource ref,
+                                                  final SAMSequenceDictionary readsDict )
+    {
+        // Write all the reference sequences to a text file as a series of 10,000-base lines.
```

Comment:
if you refer to constants in a comment you should refer to them by the name of the constant instead of referring to the value, so that the comments remain correct when values change.

I.e.   // Write all the reference sequences to a text file as a series of 10,000-base lines.